test/distributed/_sharded_tensor/ops/test_embedding.py

```diff
@@ -152,23 +152,25 @@ def test_sharded_embedding_rowwise(self):
             self._run_sharded_embedding(spec, [5, 12], 16, 22)
             self._run_sharded_embedding(spec, [5, 4], 32, 12)
             self._run_sharded_embedding(spec, [6, 7, 6], 64, 11)
-            self._run_sharded_embedding(
-                spec, [5, 12], 16, 22, max_norm=2.5, sharded_dim=0
-            )
             self._run_sharded_embedding(spec, [6, 7, 6], 64, 11, padding_idx=30)
-            self._run_sharded_embedding(
-                spec, [6, 5, 3], 26, 11, max_norm=2.0, sharded_dim=0
-            )
+            with torch.no_grad():
+                self._run_sharded_embedding(
+                    spec, [5, 12], 16, 22, max_norm=2.5, sharded_dim=0
```

Comment:
We can't expect the user to add `no_grad()` mode like this. I think the proper fix should be in `shard_parameter` itself. I think this should resolve the issue (with a copy):

```
    with torch.no_grad():                                                       
         # Reshape to get shard for this rank.                                   
         local_shard = tensor.narrow(                                            
             sharding_spec.dim,  # type: ignore[arg-type]                        
             local_metadata.shard_offsets[sharding_spec.dim],  # type: ignore[union-attr, arg-type, index]
             local_metadata.shard_sizes[sharding_spec.dim],  # type: ignore[union-attr, index]
         ).clone()                    
```

Reply:
Address the comment from reviewers. Thanks for your feedback!

Reply:
Just for learning purpose, so clone() will create a new tensor instead of creating a view of the existing tensor, right? And the update is then legit?